documentation/src/docs/recovery-procedures/table-segment-recovery.md

```diff
@@ -0,0 +1,185 @@
+		startOffset = 13135328;
+		status = 1;
+		maxRollingLength = 128767;
+		firstChunk = _system/_tables/completedTransactionsBatch-0$attributes.index.E-5-O-13131945.29bd2254-2e77-4338-a41a-f9c2af07b913;
+		lastChunk = _system/_tables/completedTransactionsBatch-0$attributes.index.E-5-O-13389479.5779f8d5-89ad-4bc6-87fe-3f99480a7271;
+		lastModified = 0;
+		firstChunkStartOffset = 13131945;
+		lastChunkStartOffset = 13389479;
+		ownerEpoch = 5;
+
+       
+        One would edit the following fields to update the metadata to reflect the chunk properties we have in step 5a).
+
+              
+                - Increment the version
+                - update the length to reflect the cumulative length of chunk(s). In the case above we would update it to 13003178.
+                - update the chunk count. in the case above it is 1, since there is just one chunk generated in step 5a.
+                - update the startOffset to 0.
+                - update the firstChunk to _system/_tables/completedTransactionsBatch-0$attributes.index.E-1-O-0.09cb6598-da98-43ad-8d9c-44ca6d523c4b
+                - update the lastChunk to the same as there is only one chunk. 
+                - update the firstChunkStartOffset to 0. Derived from "E-1-[O]-0" part in file name.
+                - update the lastChunkStartOffset to 0. (first and last chunks are same.)
+              
+
+
+     f) We have updated the attribute index segment metadata with the chunk files and their other attributes. Next step is to create the chunkMetadata itself.
+
+            Perform a put for the attribute index chunk we have in 6a.
+            
+            This is how the put/create command would look:-
+               
+              
+                table-segment put _system/_containers/storage_metadata_3 _system/_tables/completedTransactionsBatch-0$attributes.index.E-1-O-0.09cb6598-da98-43ad-8d9c-44ca6d523c4b key=_system/_tables/completedTransactionsBatch-0$attributes.index.E-1-O-0.09cb6598-da98-43ad-8d9c-44ca6d523c4b;version=1654199182158;metadataType=ChunkMetadata;name=_system/_tables/completedTransactionsBatch-0$attributes.index.E-1-O-0.09cb6598-da98-43ad-8d9c-44ca6d523c4b;length=13003178;nextChunk=null;status=1
+              
+
+             One would enter the following fields to create the metadata:-
+
+		-- key=chunk file name
+		--version= a random long number can be put in
+		--name = chunk file name
+		--length = length of the chunk file
+		--nextChunk = null (as there is only one chunk)
+             
+             Note: In case of multiple chunks, one would have to create a "put"  for each chunk after identifying the sequence of chunks based on the offsets in their name.
+
+
+
+7) Restart SegmentStore. Upon restarting one should not basically see a "DataCorruptionException" in the segment store logs. One can even perform a table-segment get-info command and check if the number of keys increase to see if   the writes after ther repair are going through.
```

Comment:
Segment Store  instead of `SegmentStore` or `segment store`.

Reply:
done.